sql/migrate/migrate.go

```diff
@@ -209,11 +262,156 @@ func DisableChecksum() PlannerOption {
+	}
+	for _, f := range files {
+		d, err := io.ReadAll(f)
+		if err != nil {
+			return err
+		}
+		if err := p.dir.WriteFile(f.Name(), d); err != nil {
+			return err
+		}
+	}
+	if p.sum {
+		sum, err := HashSum(p.dir)
+		if err != nil {
+			return err
+		}
+		return WriteSumFile(p.dir, sum)
+	}
+	return nil
+}
+
+// NewExecutor creates a new Executor with default values. // TODO(masseelch): Operator Version and other Meta
+func NewExecutor(drv Driver, dir Dir, opts ...ExecutorOption) (*Executor, error) {
+	p := &Executor{drv: drv, dir: dir}
+	for _, opt := range opts {
+		if err := opt(p); err != nil {
+			return nil, err
+		}
+	}
+	return p, nil
+}
+
+// Execute executes n missing migration files on the database
+func (e *Executor) Execute(ctx context.Context, n int) error {
+	// Don't operate with a broken migration directory. TODO(maseeelch): do not check here but let the caller check it before? Or let caller decide if to skip this flag by using some flags.
+	if err := Validate(e.dir); err != nil {
+		return fmt.Errorf("sql/migrate: execute: validate migration directory: %w", err)
+	}
+	// Check if the Driver implements RevisionReadWriter interface.
+	rrw, ok := e.drv.(RevisionReadWriter)
+	if !ok {
+		return fmt.Errorf("sql/migrate: execute: no revisions reader available")
+	}
+	// Check if the Dir implements Scanner interface.
+	sc, ok := e.dir.(Scanner)
+	if !ok {
+		return fmt.Errorf("sql/migrate: execute: no scanner available")
+	}
+	// Read all applied database revisions.
+	revisions, err := rrw.ReadRevisions(ctx)
+	if err != nil {
+		return fmt.Errorf("sql/migrate: execute: read revisions: %w", err)
+	}
+	// Select the correct migration files.
+	migrations, err := sc.Files()
+	if err != nil {
+		return fmt.Errorf("sql/migrate: execute: select migration files: %w", err)
+	}
+	// Check if the existing revisions did come from the migration directory and not from somewhere else.
+	hf, err := readHashFile(e.dir)
+	if err != nil {
+		return fmt.Errorf("sql/migrate: execute: read atlas.sum file: %w", err)
+	}
+	// For up to len(revisions) revisions the migration files must both match in order and content.
+	if len(revisions) > len(migrations) {
+		return fmt.Errorf("sql/migrate: execute: revisions and migrations mismatch: more revisions than migrations")
+	}
+	for i := range revisions {
+		m := migrations[i]
+		v, err := sc.Version(m)
+		if err != nil {
+			return fmt.Errorf("sql/migrate: execute: scan version from %q: %w", m.Name(), err)
+		}
+		d, err := sc.Desc(m)
+		if err != nil {
+			return fmt.Errorf("sql/migrate: execute: scan description from %q: %w", m.Name(), err)
+		}
+		r := revisions[i]
+		if v != r.Version || d != r.Description || hf[i].H != r.Hash { // TODO(masseelch): version / desc check necessary?
+			return fmt.Errorf("sql/migrate: execute: revisions and migrations mismatch: rev %q <> file %q", v, r.Version)
+		}
+	}
+	if len(migrations) == len(revisions) {
+		return fmt.Errorf("sql/migrate: execute: nothing to do")
```

Comment:
I think `migrate apply` command should not return an error in case there's nothing to apply.